adb_graphics/datahandler/gribdata.py

```diff
@@ -1,254 +1,153 @@
-# pylint: disable=invalid-name, too-many-public-methods, too-many-lines
-
-'''
+"""
 Classes that handle the specifics of grib files from UPP.
-'''
+"""
 
 import abc
-import datetime
-from functools import lru_cache
-from string import digits, ascii_letters
+from copy import deepcopy
+from datetime import datetime, timedelta
+from functools import cached_property
 
-from matplotlib import cm
 import numpy as np
-import xarray as xr
+from matplotlib.pyplot import get_cmap
+from pandas import to_datetime
+from uwtools.api.config import YAMLConfig
+from xarray import DataArray, Dataset, ufuncs, where
 
-from .. import conversions
-from .. import errors
-from .. import specs
-from .. import utils
+from adb_graphics import conversions, errors, specs, utils
 
-class UPPData(specs.VarSpec):
 
-    '''
+class UPPData(specs.VarSpec):
+    """
     Class provides interface for accessing field  data from UPP in
     Grib2 format.
 
     Input:
         ds:          xarray dataset from grib file
+        model:       name of the model from the image list
         short_name:  name of variable corresponding to entry in specs configuration
-
-    Keyword Arguments:
-        config:      path to a user-specified configuration file
-        model:       string describing the model type
-    '''
-
-    def __init__(self, ds, short_name, **kwargs):
-
-
-        # Parse kwargs first
-        config = kwargs.get('config', 'adb_graphics/default_specs.yml')
-        self.model = kwargs.get('model')
-        self.filetype = kwargs.get('filetype', 'prs')
-
-
-        specs.VarSpec.__init__(self, config)
-
-        self.spec = self.yml
+        spec:        full specs dictionary
+    """
+
+    def __init__(
+        self,
+        fhr: int,
+        ds: dict[str, Dataset],
+        model: str,
+        short_name: str,
+        spec: dict | YAMLConfig,
+        level: str | None = "ua",
+    ):
+        self.model = model
+        self.spec = spec
         self.short_name = short_name
-        self.level = 'ua'
-
-        self.fhr = str(kwargs['fhr'])
+        self.level = level
 
+        self.fhr = fhr
+        cf = deepcopy(self.vspec)
+        utils.set_level(level=str(level), model=self.model, spec=cf)
+        cf = utils.cfgrib_spec(cf["cfgrib"], self.model)
+        try:
+            self.vertical_coord = cf["typeOfLevel"]
+        except KeyError:
+            msg = f"typOfLevel is not a key for {short_name} at {level}. cf: {cf}"
```

Comment:
```suggestion
        key = "typeOfLevel"
        try:
            self.vertical_coord = cf[key]
        except KeyError:
            msg = f"{key} is not a key for {short_name} at {level}. cf: {cf}"
```
Alternative fix for the typo, also DRYer.